src/flashops.c

```diff
@@ -72,11 +81,60 @@ int8_t writeDataflash(MSHW* ms, unsigned int translated_addr, uint8_t val)
 	return modified;
 }
 
+/* Generate and set a random serial number valid for Mailstation
+ *
+ * Disassembly of codeflash found that the verification process for serial num
+ * is that it starts at 0x7FFC8, is 16 bytes long, each character must be
+ * ASCII '0'-'9', 'A'-'Z', 'a'-'z', or '-'
+ * In all observed Mailstations, the last character is '-', but it does
+ * not seem to be enforced in Mailstation firmware. Nevertheless, the last char
+ * is set to a '-'
+ */
+void set_rnd_ms_ser(uint8_t *buf)
```

Comment:
Instead of having to pass in the location for the serial, why not pass the start of DF to `set_rnd_ms_ser` and it can offset to `DF_SN_OFFS` itself. We then also don't need to put `DF_SN_OFFS` in the header file, unless you expect to need this offset in other places in the future.

Reply:
Refactored, made this function accept ms_ctx and modify the dataflash from there. All self contained now.

Reply:
And moved all DF related funcs to prefix with "df_"